documentation/starter-kits/CONTRIBUTING/CONTRIBUTING.md

```diff
@@ -0,0 +1,222 @@
+
+At a minimum however to submit patches (if using Git), you'll want to ensure you have:
+1. An account on the Version Control System our project uses (i.e. GitHub).
+2. The Version Control System client (i.e. Git) installed on your local machine.
+3. The ability to edit, build, and test our project on your local machine. Again, see our `README` or detailed developer guide for more details 
+
+## Our Development Process
+
+Our project integrates contributions from many people, and so we'd like to outline a process you can use to visualize how your contributions may be integrated if you provide something. 
+
+```mermaid
+flowchart TD
+    repo_proj[(Our Repository)]-->|Fork|repo_fork[(Your Forked Repository)]
+    repo_fork-->|Generate|patch(Your Patch)
+    patch-->|Submit|pr(Pull Request)
+    pr==>|Approved|repo_proj
+    pr-->|Changes Requested|repo_fork
+```
+
+### Fork our Repository
+
+Forking our repository, as opposed to directly committing to a branch is the preferred way to propose changes. 
+
+See [this GitHub guide](https://docs.github.com/en/get-started/quickstart/fork-a-repo) on forking for information specific to GitHub.com
+
+#### Find or File an Issue
+
+Make sure people are aware you're working on a patch! Check out our issue tracking system and find an open issue you'd like to work against, or alternatively file a new issue and mention you're working on a patch.
+
+### Make your Modifications
+
+Within your local development environment, this is the stage at which you'll propose your changes, and commit those changes back to version control. See the `README` or development guide for more specifics on what you'll need as prerequisites to setup your local development environment.
+
+#### Commit Messages
+
+Commit messages to version control should reference a ticket in their title / summary line:
+
+```
+Issue #248 - Show an example commit message title
+```
+
+This makes sure that tickets are updated on GitHub with references to commits that are related to them.
+
+Commit should always be atomic. Keep solutions isolated whenever possible. Filler commits such as "clean up white space" or "fix typo" should be [rebased](https://git-scm.com/docs/git-rebase) out before making a pull request (to prevent deluge of trivial commits). Please ensure your commit history is clean and meaningful!
+
+### Submit a Pull Request
+
+Pull requests are the core way our project will receive your patch contributions. Navigate to your branch on your own fork within the version control system, and submit a pull request or submit the patch text to our project. 
+
+Please make sure to provide a meaningful text description to your pull requests, whenever submitted. Use the following template:
+```
+## Issues
+- issue-XYZ
+- issue-XYZ
+## Proposed Changes
+- [ADD] ...
+- [CHANGE] ...
+- [FIX] ...
+## Testing
+- Test results available at ...
```

Comment:
This is a good question @jeffreypon - I think the intention of this block is to get the PR contributor to think about _actually testing_ the contribution before submitting the PR. Let me reword this to be less _specific_ about how to share tests, and more intentional that tests should have been done. 

Also, let me share this with @jpl-jengelke as this could be a topic to cover in the Continuous Testing guide, of _where_ to recommend placing automated test results to.